### PR TITLE
mysql doc rewrite

### PR DESCRIPTION
**Pull request by Hesperide**
Re-write of MySQL docs.

* Moves extra uncommonly needed info to a troubleshooting file.
* Matches format for Postgres and Mongo docs.
* Uses new tags `<HideInUI>`, `<details>` and `<FieldAnchor>`.
* Adds a handful of screenshots.

### CODE REVIEW

File: docs/integrations/sources/mysql.md [hunk 1]
```diff
@@ -1,91 +1,55 @@
...
-| Replicate Incremental Deletes | Yes       |                                   |
-| CDC                           | Yes       |                                   |
-| SSL Support                   | Yes       |                                   |
-| SSH Tunnel Connection         | Yes       |                                   |
-| Namespaces                    | Yes       | Enabled by default                |
-| Arrays                        | Yes       | Byte arrays are not supported yet |
+The contents below include a 'Quick Start' guide, advanced setup steps, and reference information (data type mapping, and changelogs).
 
-The MySQL source does not alter the schema present in your database. Depending on the destination connected to this source, however, the schema may be altered. See the destination's documentation for more details.
+![Airbyte MySQL Connection](./mysql/assets/airbyte_mysql_source.png)
```
Comment (Hesperide) [inline docs/integrations/sources/mysql.md:10]:
```suggestion
![Airbyte MySQL Connection](https://raw.githubusercontent.com/airbytehq/airbyte/3a9264666b7b9b9d10ef8d174b8454a6c7e57560/docs/integrations/sources/mysql/assets/airbyte_mysql_source.png)
```

File: docs/integrations/sources/mysql.md [hunk 2]
```diff
@@ -1,91 +1,55 @@
...
+1. Create a dedicated read-only MySQL user with permissions for replicating data
+2. Create a new MySQL source in the Airbyte UI using CDC logical replication
+3. (Airbyte Cloud Only) Allow inbound traffic from Airbyte IPs
 
-Some users reported that they could not connect to Amazon RDS MySQL or MariaDB. This can be diagnosed with the error message: `Cannot create a PoolableConnectionFactory`.
-To solve this issue add `enabledTLSProtocols=TLSv1.2` in the JDBC parameters.
+Once this is complete, you will be able to select MySQL as a source for replicating data.
 
-Another error that users have reported when trying to connect to Amazon RDS MySQL is `Error: HikariPool-1 - Connection is not available, request timed out after 30001ms.`. Many times this is can be due to the VPC not allowing public traffic, however, we recommend going through [this AWS troubleshooting checklist](https://aws.amazon.com/premiumsupport/knowledge-center/rds-cannot-connect/) to the correct permissions/settings have been granted to allow connection to your database.
+<FieldAnchor field="username,password">
```
Comment (lmossman) [inline docs/integrations/sources/mysql.md:22]:
```suggestion
<FieldAnchor field="username,password">
```

I'm still investigating why this is the case, but the `#### Step 1:...` heading doesn't render properly in the Airbyte UI if there is an empty line between it and the <FieldAnchor> tag. 

I don't want to block this change on me investigating the bug so let's just get rid of the empty line here for now

**Review by lmossman — APPROVED** [review]
Left some suggestions but nothing truly blocking. LGTM once those are addressed

**Review by akashkulk — APPROVED** [review]
Nice! Is the troubleshooting.md file linked from anywhere? Couldn't find the link to it from the main mysql set up page.

File: docs/integrations/sources/mysql.md [hunk 3]
```diff
@@ -1,91 +1,55 @@
...
-| Replicate Incremental Deletes | Yes       |                                   |
-| CDC                           | Yes       |                                   |
-| SSL Support                   | Yes       |                                   |
-| SSH Tunnel Connection         | Yes       |                                   |
-| Namespaces                    | Yes       | Enabled by default                |
-| Arrays                        | Yes       | Byte arrays are not supported yet |
+The contents below include a 'Quick Start' guide, advanced setup steps, and reference information (data type mapping, and changelogs).
 
-The MySQL source does not alter the schema present in your database. Depending on the destination connected to this source, however, the schema may be altered. See the destination's documentation for more details.
+![Airbyte MySQL Connection](https://raw.githubusercontent.com/airbytehq/airbyte/3a9264666b7b9b9d10ef8d174b8454a6c7e57560/docs/integrations/sources/mysql/assets/airbyte_mysql_source.png)
```
[the remark this answers is not in the document]
Comment (Hesperide) [reply docs/integrations/sources/mysql.md:10]: Common practice to include a screenshot of what something looks like when you are done at the beginning of instructions :)

File: docs/integrations/sources/mysql.md [hunk 4]
```diff
@@ -221,44 +207,8 @@ Any database or table encoding combination of charset and collation is supported
...
-
-If you have connections with MySQL Source using _Logical Replication (CDC)_ method, run this SQL:
-
-```sql
-update public.actor set configuration =jsonb_set(configuration, '{replication_method}', '{"method": "CDC"}', true)
-WHERE actor_definition_id ='435bb9a5-7887-4809-aa58-28c27df0d7ad' AND (configuration->>'replication_method' = 'CDC');
-```
+</details>
 
 ## Changelog
```
[the remark this answers is not in the document]
Comment (Hesperide) [reply docs/integrations/sources/mysql.md:213]: I would advocate for this, but would prefer to do this all at once for all connectors.

File: docs/integrations/sources/mysql.md [hunk 5]
```diff
@@ -101,74 +65,91 @@ binlog_expire_logs_seconds  = 864000
...
 
-The MySQl connector may need some time to start processing the data in the CDC mode in the following scenarios:
+<HideInUI>
 
-- When the connection is set up for the first time and a snapshot is needed
-- When the connector has a lot of change logs to process
+![Create an Airbyte source](https://github.com/airbytehq/airbyte/blob/c078e8ed6703020a584d9362efa5665fbe8db77f/docs/integrations/sources/postgres/assets/airbyte_source_selection.png?raw=true)
 
-The connector waits for the default initial wait time of 5 minutes (300 seconds). Setting the parameter to a longer duration will result in slower syncs, while setting it to a shorter duration may cause the connector to not have enough time to create the initial snapshot or read through the change logs. The valid range is 300 seconds to 1200 seconds.
+</HideInUI>
```
[the remark this answers is not in the document]
Comment (Hesperide) [reply docs/integrations/sources/mysql.md:82]: I see where you are coming from (and wouldn't be strongly opposed), but we also have a step 4. And I don't want us to have to worry about Step 1 -> 2 -> 4 (skipping 3).

Maybe something we can look at improving in the future.

**Pull request merged**